### ISSUE TITLE
Alert Sound on Copy

### ISSUE THREAD

**Issue by imericxu** [p0deje/Maccy#748]
### Before Submitting Your Bug Report

- [X] I have verified that there isn't already an issue reporting the same bug to prevent duplication.
- [X] I have seen the [FAQ](https://github.com/p0deje/Maccy?tab=readme-ov-file#faq).

### Maccy Version (see 'About' window)

0.30.0

### macOS Version

14.4

### Maccy Settings

```Shell
{
    "KeyboardShortcuts_delete" = "{\\"carbonModifiers\\":2048,\\"carbonKeyCode\\":51}";
    "KeyboardShortcuts_pin" = "{\\"carbonModifiers\\":2048,\\"carbonKeyCode\\":35}";
    "KeyboardShortcuts_popup" = "{\\"carbonKeyCode\\":8,\\"carbonModifiers\\":768}";
    "LaunchAtLogin__hasMigrated" = 1;
    "NSStatusItem Preferred Position Item-0" = 810;
    "NSWindow Frame com.sindresorhus.Settings.FrameAutosaveName" = "734 251 542 380 0 0 1792 1095 ";
    clearSystemClipboard = 1;
    enabledPasteboardTypes =     (
        "public.png",
        "public.html",
        "public.utf8-plain-text",
        "public.rtf",
        "public.tiff",
        "public.file-url"
    );
    historySize = 50;
    ignoreEvents = 0;
    ignoredPasteboardTypes =     (
        "net.antelle.keeweb",
        "com.typeit4me.clipping",
        "de.petermaurer.TransientPasteboardType",
        "Pasteboard generator type",
        "com.agilebits.onepassword"
    );
    imageMaxHeight = 48;
    lastReviewRequestedAt = 1708468975;
    menuIcon = paperclip;
    migrations =     {
        "2020-04-25-allow-custom-ignored-types" = 1;
        "2020-06-19-use-keyboardshortcuts" = 1;
        "2020-09-01-ignore-keeweb" = 1;
        "2021-02-20-allow-to-customize-supported-types" = 1;
        "2021-06-28-add-title-to-history-item" = 1;
        "2021-10-16-remove-dynamic-pasteboard-types" = 1;
        "2022-08-01-rename-suppress-clear-alert" = 1;
        "2022-11-14-add-html-rtf-to-supported-types" = 1;
        "2023-01-22-add-regexp-search-mode" = 1;
    };
    numberOfUsages = 168;
    pasteByDefault = 0;
    popupPosition = statusItem;
    previewDelay = 800;
    removeFormattingByDefault = 1;
    searchMode = fuzzy;
    showRecentCopyInMenuBar = 0;
}
```

### Description

When copying a link using the ⌘ keyboard shortcut, macOS plays the alert sound. I can’t tell in what contexts this occurs, because it doesn’t happen in some apps, and even in those apps, it depends on where you’re focused.
Likely this (Boop): <https://youtu.be/SPDsDn7Terw?si=GsWNJj4I8aTgeDsX>

### Steps to Reproduce

1. Make sure volume is on
2. ⌘ + ⇧ + J to open the menu
3. ⌘ + 1 (or another number) to copy/paste
4. Listen for alert sound
5. Repeat steps 2–4 in different contexts

**Comment by p0deje** [comment]
You can disable sounds in "Notifications & sounds" in Maccy general preferences.

**Comment by Imericxu** [comment]
I think you misunderstood me. The macOS alerts are separate from the Maccy notification sounds. The alert plays regardless of whether Maccy has notifications on.

**Comment by p0deje** [comment]
Ah, can you try restarting Maccy? Sometimes it might have issues focusing the window back and when it attempts to paste by pressing CMD+V, the alert sound is played. It means that pasting failed to happen. There is not much I can do in Maccy about it unfortunately.

**Comment by p0deje** [comment]
Personally, I have "Paste automatically" disabled in Maccy preferences. Only if I want to paste right away (e.g. I'm typing something in browser), I would select item with Option pressed to paste. Otherwise, I would just let Maccy copy item to the system clipboard so I could later paste it as I wish.

**Comment by p0deje** [comment]
Please ignore my comments, this is actually a different problem. I'll investigate more.

**Comment by geiras** [comment]
Hey. UI/UX designer here. In an attempt to identify the source of a mysterious sound, I discovered a terminal command that lists apps using the sound card. I found Maccy among them and noticed that when copying text with the Command+C shortcut, it produced an writing-like sound.

An option to disable this sound effect within Mackie's settings would be preferable. I tried closing the app, I had difficulty doing so without re-enabling the footer. Once done, however, the alert sound ceased until Maccy was reopened.

Upon further investigation, I found a "notifications and sounds" help link within Maccy settings that took me to MacOS System Preferences. By disabling this feature, the issue was resolved and no more sound plays when copying someting. 

However, I believe this option should be more prominently displayed alongside other settings for easier access.
Although the operating system may wish to maintain control over such features, incorporating this option directly into Maccy preferences would streamline user experience and allow for better management of settings related to the app. Ideally, these preferences would also sync with the overall system settings if possible.

**Comment by stephanknotek** [comment]
I just got Maccy and I immediately started looking for a way to mute this sound in Maccy's preferences, but I found no such option, and found my way here. I'd like to also advocate for this option because I need my Mac's sound on to be alerted for emails if I'm working near my desk but not watching for notifications, among other things, but Maccy's sound is rather distracting at work, especially with the amount of copying and pasting that I do as part of my workflow.

**Comment by yathu-arulnanthy-gtg** [comment]
Yes totally agree, please there needs to be an option to turn the sound off.

**Comment by imericxu** [comment]
The comments from geiras and after aren’t relevant to this issue. This bug is for the macOS system alert that plays unexpectedly on using a shortcut, not the actual Maccy notifications.

**Comment by intothevoid0** [comment]
I just started using Maccy (as a replacement for CopyClip) and I love it so far. I hope I'm posting in the correct issue thread, and if not, I apologize.

I believe I'm experiencing the same issue. I've installed on my two primary devices - iMac (Sonoma 14.1.1) and MacBook Air (Sonoma 14.4.1). On the iMac, I have no issues when opening the popup and copying an item with 'COMMAND (⌘)' + 'n'. On my MacBook Air, it makes the "not allowed" sound effect when selecting an item in multiple applications. As an example, when Finder is active, and I use '⌘1', I believe it is attempting to trigger View > as Icons. For Chrome, the same thing but '⌘ 1-9' for changing to a different tab.

I'm not sure what changed in macOS, but I hope you find this helpful.

**Comment by andrewbraxton** [comment]
Experiencing exactly what @intothevoid0 describes on Sonoma 14.5.

**Comment by janosh** [comment]
also randomly started getting a paper-crumpling sound on copy on a new MacBook running Sonoma 14.5. didn't realize it was coming from Maccy and was quite hard to trace here. definitely needs to be made an opt in feature, i never enabled this explicitly.

**Comment by imericxu** [comment]
@janosh The paper crumpling (scribbling) sound is a notification and can be turned off in System Preferences.

**Comment by janosh** [comment]
yeah i did. just took me a while to figure out which app it was coming from

**Comment by intothevoid0** [comment]
I'm still running Maccy v0.31.0 on my MacBook Air. I tried reading the release notes, but they went over my head. Can someone please explain the details of the update to v1.0.0 and let me know if the sound issue was addressed?

**Comment by p0deje** [comment]
> I'm still running Maccy v0.31.0 on my MacBook Air. I tried reading the release notes, but they went over my head. Can someone please explain the details of the update to v1.0.0 and let me know if the sound issue was addressed?

@intothevoid0 It's not fixed in 1.0, but it will be fixed in 2.0.

**Comment by intothevoid0** [comment]
@p0deje, thanks for getting back to me; I'm glad to hear it! I primarily use my iMac for work and it’s not affected by the sound issue — I was just curious.

When you have a moment, could you please elaborate on the following info from the release notes?

> Please note that older versions of Maccy (0.x) will not work correctly after the upgrade.
> - Enabled persistent history tracking in storage. This forces read-only mode in Maccy 0.x.

**Comment by p0deje** [comment]
> When you have a moment, could you please elaborate on the following info from the release notes?

What exactly to elaborate? Once you upgrade to 1.0, you cannot use 0.x version anymore.

**Comment by p0deje** [comment]
Fixed in 2.0
